### PR TITLE
[e2e] do not refresh on destroy

### PR DESCRIPTION
**Pull request by pducolin**
### What does this PR do?

Improve new e2e clean up task. by trying first to destroy the stack, than destroy with refresh on failure, and eventually remove the local stack with force.  

### Motivation

Refreshing on destroy can trigger resources update, that is unwanted. 

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: tasks/new_e2e_tests.py [hunk 1]
```diff
@@ -205,20 +205,32 @@ def _get_existing_stacks(ctx: Context) -> List[str]:
...
+    # with resources removed by agent-sandbox clean up job
     with ctx.cd(tempfile.gettempdir()):
-        ctx.run(f"pulumi destroy --stack {stack_name} -r --yes --remove --skip-preview", pty=True)
+        ret = ctx.run(
+            f"PULUMI_SKIP_UPDATE_CHECK=true pulumi destroy --stack {stack_name} --yes --remove --skip-preview",
+            pty=True,
+            warn=True,
+        )
+        if ret is not None and ret.exited != 0:
+            # run with refresh on first destroy attempt failure
```
Comment (Pythyu) [inline tasks/new_e2e_tests.py:217]: Question❓ 
I don't see anything added to refresh aside from trying to re-destroy again, is it normal ?

**Review by Pythyu — APPROVED** [review]
Aside from my comment, LGTM 👍

File: tasks/new_e2e_tests.py (hunk 1 above)
Comment (pducolin) [reply tasks/new_e2e_tests.py:217]: True it is super tiny, it is the `-r` in the second command triggering the `refresh`

File: tasks/new_e2e_tests.py [hunk 2]
```diff
@@ -205,20 +205,32 @@ def _get_existing_stacks(ctx: Context) -> List[str]:
...
+            ctx.run(
+                f"PULUMI_SKIP_UPDATE_CHECK=true pulumi destroy --stack {stack_name} -r --yes --remove --skip-preview",
```
Comment (pducolin) [inline tasks/new_e2e_tests.py:219]: `-r` here triggeres destroy with refresh. Can change it to `--refresh="true"` to make it more readable

**Pull request merged**